KeyValue.cpp

```diff
@@ -77,7 +77,11 @@ KeyValueErrorCode ReadQuotedString(const char*& str, kvString_t& inset)
 
 	inset.string = const_cast<char*>(str);
 
-	for (char c = *str; c && c != STRING_CONTAINER; c = *++str);
+	bool escaped = true;
+	for (char c = *str; c && (c != STRING_CONTAINER || escaped); c = *++str)
+	{
+		escaped = c == '\\';
+	}
```

Comment:
Given that the initial value of `escaped` is `true`, this code will break for the string `""`. We would jump over the first quotation mark, and the first character `c` will be equal to the last quotation mark `"`. Since `escaped` is true this will cause the for loop to continue past the end and read out of bounds. Let's toss out the extra state and get rid of `bool escaped`. Instead, since we know the very first character must always be a `"`, we can safely backtrack one character and check if it's an escaped character. Using `str[-1] == '\\'` would be equivalent and use less state. Alternatively, storing the last character would also be fine, as that's one less dereference.

Reply:
yeah it should be false, i forgot to change it when changed the behavior

Reply:
Just had a realization `"\\"` is also a valid string! Seems like we will need a variable for if it's escaped after all! :P

Also, the `[]` operator is just math! You can do `str[0]` or `0[str]`, and it's the same as `*(str + 0)` or `*(0 + str)`.